### PR TITLE
Med 55 integrate api for local doctor profile

### PR DESCRIPTION
**Pull request by skazzp**
- Added Profile Page integration, "edit profile" option;
- Added Google Auth integration;
- Added protected routing, checks for verification, filled profile etc.
- Added basic layout with navigation for protected routes;

**PR checklist:**

**1.1 Common**
- [x] 1.1.2 no any, should be also added to eslint rules
- [x] 1.1.1 types for input and output params
- [x] 1.1.3 try/catch for any async function
- [x] 1.1.4 no magic numbers
- [x] 1.1.5 compare only with constants not with strings (instead of user === ‘admin’, better user === roles.admin)
- [x] 1.1.6 no ternary operator inside ternary operator (bad example: question ? first: second ? third: fourth)
- [x] 1.1.7 avoid similar types with duplicating by generics
- [x] 1.1.8 no console.log in pr
- [x] 1.1.9 .env file should be in .gitignore
- [x] 1.1.10 delete commented code if it's not part of documentation
- [x] 1.1.11 no links in the code, env links should be in env file (for example: server url),
- [x] constant links (for example default avatar URL) should be in constant file
- [x] 1.1.12 constants and function names should be lowercase.
- [x] 1.1.13 no dates format in the code (like "dd/MM/yyyy”), move it in global constant variable
- [x] 1.1.14 import rules. imports should come in a specific order: node modules, absolute path, relative path
- [x] 1.1.15 strings should be in the constant variable. Example: instead of ‘15 3 * * *’, const EACH_DAY_15h_15min = ‘15 3 * * *’

 **1.3 Frontend|Mobile**
- [x] 1.3.1 split component and logic (move logic to custom hook file)
- [x] 1.3.2 colors, font size, and font name should be in the theme or in the constants
- [x] 1.3.3 no text in the components, use i18n approach, even if you have only one language for now
- [x] 1.3.4 inline styles prohibited
- [x] 1.3.5 import should be absolute. instead of ../../../components/myComponent should be components/myComponent

Profile after registration by Google (all routes except the profile are blocked)
image

Filled profile (role/speciality only shown when role is not set; email change is blocked)
image

Login page with Google button
image

### CODE REVIEW

File: apps/medon-fe/src/components/LoginForm/style.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { theme } from 'styles/theme';
```
Comment (Mati-Pereira) [inline apps/medon-fe/src/components/LoginForm/style.tsx:2]: order of imports

File: apps/medon-fe/src/components/ProfileForm/index.tsx [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
-  DEFAULT_TIMEZONE,
-  timezoneOptions,
-} from 'utils/timezones/timezoneOptions';
+import { ROLES, ROLE_OPTIONS } from 'utils/constants/roles';
+import { timezoneOptions } from 'utils/timezones/timezoneOptions';
 import { formFields } from 'utils/constants/userFormFields';
 import { countryOptions } from 'utils/countries/countryOptions';
 import { profileFormSchema } from 'validation/profileFormSchema';
+import { useAppDispatch, useAppSelector } from 'redux/hooks';
+import { getUserSelector } from 'redux/features/userSlice/userSelectors';
```
Comment (Mati-Pereira) [inline apps/medon-fe/src/components/ProfileForm/index.tsx]: sorting

File: apps/medon-fe/src/pages/RegistrationPage/index.tsx [hunk 3]
```diff
@@ -1,12 +1,13 @@
-import RegistrationForm from 'components/RegistrationForm';
+import { useState } from 'react';
 import { useTranslation } from 'react-i18next';
-import logo from 'assets/images/logo.svg';
+import { toast } from 'react-toastify';
+import dayjs from 'dayjs';
+import RegistrationForm from 'components/RegistrationForm';
 import RegistrationConfirmation from 'components/RegistrationConfirmation';
```
Comment (Mati-Pereira) [inline apps/medon-fe/src/pages/RegistrationPage/index.tsx:6]: order

**Review by Mati-Pereira — CHANGES_REQUESTED** [review]
that's all

File: apps/medon-fe/src/pages/RegistrationPage/index.tsx (hunk 3 above)
Comment (skazzp) [reply apps/medon-fe/src/pages/RegistrationPage/index.tsx:6]: done

File: apps/medon-fe/src/components/ProfileForm/index.tsx (hunk 2 above)
Comment (skazzp) [reply apps/medon-fe/src/components/ProfileForm/index.tsx]: done

File: apps/medon-fe/src/components/LoginForm/style.tsx (hunk 1 above)
Comment (skazzp) [reply apps/medon-fe/src/components/LoginForm/style.tsx:2]: done

**Review by Mati-Pereira — APPROVED** [review]

**Review by SoulOfDarknes — APPROVED** [review]

File: apps/medon-fe/src/components/LoginForm/index.tsx [hunk 4]
```diff
@@ -37,12 +39,17 @@ const LoginForm: FC = () => {
...
     try {
+      dispatch(userApi.util.invalidateTags(['user']));
       await login(formData).unwrap();
     } catch (error) {
       toast.error(t('login.error-msg'), toastConfig);
     }
   };
 
+  const handleGoogleClick = () => {
+    window.location.href = `${process.env.NX_API_URL}${process.env.NX_API_GOOGLE_ROUTE}`;
```
Comment (lyudad) [inline apps/medon-fe/src/components/LoginForm/index.tsx]: find other way window better no to use

Comment (skazzp) [reply apps/medon-fe/src/components/LoginForm/index.tsx]: removed, button changed to antD button

**Pull request merged**